components/root/root.jsx

```diff
@@ -235,6 +238,8 @@ export default class Root extends React.Component {
         if (props.location.pathname === '/') {
             if (this.props.noAccounts) {
                 this.props.history.push('/signup_user_complete');
+            } else if (props.showTermsOfService) {
+                this.props.history.push('/terms_of_service');
```

Comment:
Actually, this seems like the best, top-level place to do this redirect: do we need handling in the login controller and the logged in component?

Reply:
We could, but then we would need a method to check if the user has opened a `login required route`. 
Since the current behaviour is that we don't redirect for routes like: `/error`, `/help` etc.